src/error/mod.rs

```diff
@@ -659,12 +659,9 @@ impl<F: ErrorFormatter> Error<F> {
             let mut suggestions = vec![];
             if suggested_trailing_arg {
                 let mut styled_suggestion = StyledStr::new();
-                styled_suggestion.none("If you tried to supply '");
-                styled_suggestion.warning(&arg);
-                styled_suggestion.none("' as a value rather than a flag, use '");
-                styled_suggestion.good("-- ");
-                styled_suggestion.good(&arg);
-                styled_suggestion.none("'");
+                styled_suggestion.none("There is an flag with the same name, try to remove `");
+                styled_suggestion.good("--");
+                styled_suggestion.none("`");
                 suggestions.push(styled_suggestion);
```

Comment:
As highlighted by my previous comments, this isn't the right way to fix this issue.  In a lot of cases, a scenario is being detected and this is the text for that scenario. 

What I would recommend is creating a reproduction case for this issue, committing that, and then investigate why it shares a message with these other cases, and then create a fix for that.  That'll be in the `src/parser/parser.rs` code and might require some new logic here, unsure.